src/planscape/forsys/generate_projects_for_a_single_scenario.R

```diff
@@ -0,0 +1,69 @@
+suppressMessages({
+  library(dplyr)
+  library(sf)
+  library(forsys)
+  library(patchmax)
+})
+
+generate_projects_for_a_single_scenario <- function(forsys_input_data,
+                                                    priorities,
+                                                    priority_weights,
+                                                    stand_id_field,
+                                                    proj_id_field,
+                                                    stand_area_field,
+                                                    stand_cost_field,
+                                                    geo_wkt_field) {
+
+  scenario_output_fields <- c(priorities, stand_area_field, stand_cost_field)
+
+  priority_impact_fields <- paste(priorities, "_PCP", sep="")
+
+  # Appends spm and pcp data columns for each priority.
+  forsys_input_data <- forsys_input_data %>% 
+                forsys::calculate_spm(fields = priorities) %>%
+                forsys::calculate_pcp(fields = priorities)
+  # Appends a preset_priority column, which contains total impact score values.
+  forsys_input_data <- forsys_input_data %>%
+                forsys::combine_priorities(
+                  fields = priority_impact_fields, 
+                  weights = priority_weights, 
+                  new_field = 'preset_priority')
```

Comment:
It can be named anything. I called it preset_priority because that's what it was called in an example provided by Michelle.
I'll rename it to "weighted_priorities".